generator/internal/language/templates/rust/crate/src/transport.rs.mustache

```diff
@@ -56,10 +56,21 @@ impl crate::traits::{{NameToPascal}} for {{NameToPascal}} {
     ) -> Result<{{OutputTypeName}}> {
         let options = options.set_default_idempotency(reqwest::Method::{{HTTPMethod}}.is_idempotent());
         let builder = self.inner.builder(
-            reqwest::Method::{{HTTPMethod}}, format!("{{HTTPPathFmt}}"
-               {{#HTTPPathArgs}}
-               , {{{.}}}
-               {{/HTTPPathArgs}}
+            reqwest::Method::{{HTTPMethod}},
+            format!("{{#HTTPPathInfo.PathTemplate}}{{#Literal}}/{{Value}}{{/Literal}}{{#FieldPath}}/{}{{/FieldPath}}{{#Verb}}:{{Value}}{{/Verb}}{{/HTTPPathInfo.PathTemplate}}"
+            {{#HTTPPathInfo.PathTemplate}}{{#FieldPath}},
+                req
+                {{#Components}}
+                    {{#Reference}}
+                        {{#Optional}}
+                            .{{Codec.FieldName}}.as_ref().ok_or_else(|| gax::error::Error::other("field \"{{Codec.FieldName}}\" is required in a request, but was missing"))?
```

Comment:
We can use a helper function to make this code smaller:

```suggestion
                            .{{Codec.FieldName}}
                                .as_ref()
                                .ok_or_else(|| gax::path_parameter::missing("{{Codec.FieldName}}")?
```

Reply:
Nice, I'll add that!

Reply:
I couldn't figure out how to add the function directly under `gax::path_parameter`, so I made it `gax::path_parameter::errors::missing`.